### DISCUSSION TITLE
formFactor simulation device

### DISCUSSION THREAD

**Discussion by student513** [GoogleChrome/lighthouse-ci#556] [Q&A]
I want to know the simulation environment according to `formFactor`.
We found that `formFactor=mobile` is simulated on a device called `andriod 7.0 motor G4`.
If it is `formFactor=desktop`, I only know it runs on the desktop.
Where can I find information about exactly what environment it runs in?
I would like to know more information such as OS, CPU, browser version, etc.

These are my configuration.
Case `formFactor=mobile`
```ts
--settings.chromeFlags="--no-sandbox"
--settings.formFactor="mobile"
--settings.screenEmulation.mobile
```

Case `formFactor=desktop`
```ts
--settings.chromeFlags="--no-sandbox"
--settings.formFactor="desktop"
--settings.screenEmulation.disabled
```

envirment
lhci version: 7.0.0
OS: CentOS Linux release 7.9.2009 (Core)

**Comment by patrickhulce** [comment]
> Where can I find information about exactly what environment it runs in?

You'll probably want to read the [Lighthouse emulation docs](https://github.com/GoogleChrome/lighthouse/blob/master/docs/emulation.md). It doesn't control what you think it does :)


> I would like to know more information such as OS, CPU, browser version, etc.

Just to clarify, setting anything here doesn't mean a different device is somehow used instead of wherever you're running Lighthouse, the flags control the *emulation* or *simulation* state (i.e. faking the user agent string, artificially slowing down the connection, etc). The OS, CPU, and browser version that is used to run the test will never change just because of these flags. Those values are determined by your host environment.

**Comment by student513** [comment]
@patrickhulce 
Thanks for your answer.
I thought that emulation environment can be set through `formFactor`. However, according to the document you provided, it seems to be an option to inform the environment in which `lighthouse` If I run `lighthouse` through my phone, I have to set it as `formfactor=mobile` and hand over `object` parameter about screen size to `screen Emulation.mobile`. Am I right to understand?

> Those values are determined by your host environment.

And additionally, is the lighthouse's score dependent on the host environment in which the analysis runs?

**Reply by patrickhulce** [reply]
Yes, `formFactor` doesn't control the emulation itself, it's letting Lighthouse know what type of device you're running on and you must set other parameters to define exactly what emulation settings you'd like to use.

> And additionally, is the lighthouse's score dependent on the host environment in which the analysis runs?

Correct, you might be interested in [the variability docs](https://github.com/GoogleChrome/lighthouse/blob/master/docs/variability.md).

**Reply by student513** [reply]
I appreciate your help, thanks 🙏

**Comment by student513** [comment]
There is one thing I want to point out for sure. So does `Radiobutton`, who chooses `desktop/mobile` from `lighthouse extension`, also choose formFactor option?

![image](https://user-images.githubusercontent.com/50268619/108932054-12552100-768c-11eb-8aee-dabbc35f22bc.png)

**Reply by patrickhulce** [reply]
It does set the form factor yes, but it also sets [many other properties of emulation and throttling](https://github.com/GoogleChrome/lighthouse/blob/090a1a3f082d8d487dca2d6278115e19904336a6/lighthouse-core/config/desktop-config.js#L14-L17).
